src/asm/parser.y

```diff
@@ -1026,15 +1026,15 @@ align_spec	: uconst {
 				$$.alignOfs = 0;
 			}
 		}
-		| uconst T_COMMA uconst {
+		| uconst T_COMMA const {
 			if ($1 > 16) {
 				error("Alignment must be between 0 and 16, not %u\n", $1);
-			} else if ($3 >= 1 << $1) {
-				error("Alignment offset (%u) must be smaller than alignment size (%u)\n",
+			} else if ($3 <= -(1 << $1) || $3 >= 1 << $1) {
+				error("Alignment offset (%d) must be smaller than alignment size (%u)\n",
 					$3, 1 << $1);
```

Comment:
```suggestion
			} else if ($3 >= 1 << $1) {
				error("Alignment offset (%d) must be smaller than alignment size (%u)\n",
					$3, 1 << $1);
			} else if ($3 <= -(1 << $1)) {
				error("Alignment offset (%d) must be larger than the alignment size (-%d)\n",
					$3, 1 << $1);
```

Otherwise people may get confused by out-of-range negatives, I think.

Reply:
I'd be more confused by an error message citing a negative alignment size. How about this: "`Alignment offset (%d, equivalent to %d) must be smaller than the alignment size (-%d)`".

Reply:
I wasn't sure whether we should go with that, but yeah let's.